ouroboros-consensus/src/Ouroboros/Consensus/Node/DbLock.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Ouroboros.Consensus.Node.DbLock (
+    DbLocked (..)
+  , withLockDB
+  , dbLockFile
+  ) where
+
+import           Control.Monad.Class.MonadAsync
+import           Control.Monad.Class.MonadTimer
+import           Data.Text
+import qualified Data.Time.Clock as Time
+import           System.FileLock
+
+import           Ouroboros.Consensus.Storage.FS.API
+import           Ouroboros.Consensus.Storage.FS.API.Types
+import           Ouroboros.Consensus.Util.IOLike
+
+-- We use an empty file as a lock of the db. Some systems may delete the empty
```

Comment:
Make it a Haddock docstring and explain why we do this, e.g., by replacing the first sentence with:
> We use an empty file as a lock of the db so that the database cannot be opened by more than one process.